internal/impl/azure/output_adls.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azure
+
+import (
+	"context"
+	"fmt"
+
+	dlservice "github.com/Azure/azure-sdk-for-go/sdk/storage/azdatalake/service"
+	"github.com/redpanda-data/benthos/v4/public/service"
+)
+
+const (
+	// Azure Data Lake Storage Output Fields
+	adlsFieldFilesystem = "filesystem"
+	adlsFieldPath       = "path"
+)
+
+type adlsConfig struct {
+	client     *dlservice.Client
+	path       *service.InterpolatedString
+	filesystem *service.InterpolatedString
+}
+
+func init() {
+	err := service.RegisterOutput("azure_data_lake_gen2", adlsSpec(),
+		func(conf *service.ParsedConfig, mgr *service.Resources) (out service.Output, mif int, err error) {
+			var pConf *adlsConfig
+			if pConf, err = adlsConfigFromParsed(conf); err != nil {
+				return
+			}
+			if mif, err = conf.FieldMaxInFlight(); err != nil {
+				return
+			}
+			if out, err = newAzureADLSWriter(pConf, mgr.Logger()); err != nil {
+				return
+			}
+			return
+		})
+	if err != nil {
+		panic(err)
+	}
+}
+
+func adlsConfigFromParsed(pConf *service.ParsedConfig) (*adlsConfig, error) {
+	var conf adlsConfig
+	var err error
+	conf.filesystem, err = pConf.FieldInterpolatedString(adlsFieldFilesystem)
+	if err != nil {
+		return nil, err
+	}
+	conf.path, err = pConf.FieldInterpolatedString(adlsFieldPath)
+	if err != nil {
+		return nil, err
+	}
+	var isFilesystemSASToken bool
+	conf.client, isFilesystemSASToken, err = adlsClientFromParsed(pConf, conf.filesystem)
+	if err != nil {
+		return nil, err
+	}
+	if isFilesystemSASToken {
+		conf.filesystem, _ = service.NewInterpolatedString("")
```

Comment:
Shouldn't this be `service.NewInterpolatedString(adlsFieldFilesystem)` and have error handling like the rest?

Reply:
I cargo culted this from the `azure_blob_storage` output.

https://github.com/redpanda-data/connect/blob/9e67ce7a8b85eb150cab6afa81eb9db0dc06f541/internal/impl/azure/output_blob_storage.go#L56-L57

Reply:
Thanks for looking into it! I forgot about that... If it works the same way for Data Lake, then it's fine.